### DISCUSSION TITLE
Boot stage 2

### DISCUSSION THREAD

**Discussion by Ing-Dom** [earlephilhower/arduino-pico#272] [Q&A]
whats is the correct selection for a W25Q128 flash ?
Generic SPI /4 ?

**Comment by earlephilhower** [comment]
I suggest just trying the possibilities.  Due to wiring/capacitance issues, it's not always clear-cut.  It won't damage your board, and if it crashes you just need to hold down BOOTSEL and re-upload another version.

Generic/4 is called the "potato" flash interface in the SDK because it will work with *anything* but is really slow.

The /4 or /2 means "clk/2" or "clk/4", so `/2` would be preferred as it's 2x faster than the `/4`.

**Reply by Ing-Dom** [reply]
ok thanks.
I tried W25Q080 / 2 which works with a W25Q128JVSQ.
Seems that "W25Q080" stands for the whole W25Q80, 16, 32, 64, 128 familiy... but why there is a "0" between the Q and the 80 ?
that confused me at the beginning.

**Reply by NuclearPhoenixx** [reply]
The W25Q080 is a 1MB chip package, so the zero is correct here as it exists. However, I'm not sure how you're supposed to use a 1MB chip when the minimum selectable flash size in the menu is 2MB.

**Reply by earlephilhower** [reply]
Are people building and shipping boards with only 1MB flash on them?  Seems like a false economy.  FWIW the names of BOOT2.S files are from the Pico SDK.

In any case, if they are shipping 1MB flash boards, we can add a 1MB option to the `Generic RP2040` board item itself.  You will have issues with wraparound otherwise (although if you don't use EEPROM or LittleFS and keep the binary under 1MB in size it *may* work).

**Reply by NuclearPhoenixx** [reply]
I don't know if people use 1MB chips. However, in the boot stage menu it clearly says "W25Q080..." which I understand is a 1MB flash chip (...080...) that you can select. So why have that and then not have a 1MB flash size option? The IS25LP080 option in the same menu also looks like a 1MB flash chip, so I'm pretty confused. I'm probably having a slight misunderstanding here.

**Reply by earlephilhower** [reply]
Like I said, I just imported the names from the SDK.  I would guess that there are slightly different timings and/or protocol quirks as manufacturers and even flash sizes change, and they picked the LCD as the name.  Nothing to understand other than the name of the BOOT.S is going to reference a specific set of timings and quirks for a particular flash device series.  (Note that the BOOT2.S has no correlation to actual flash size in any way...that's a SW abstraction in the app/core itself.)

As for why no 1MB options in the menus...nobody has ever asked for it.  😄   If you have a need we can add it, OTW I'd just leave it out since we already have tons of menu items already...

**Reply by NuclearPhoenixx** [reply]
Ok thanks, I see. Sorry I was just really confused that you can select these flash chips, but not have the flash size options for them. I don't think anyone's going to use the 1MB version anyways, like you said these chips are so cheap and tiny it really doesn't matter economically to go for less flash lol
